src/instance.rs

```diff
@@ -290,7 +290,7 @@ where
         unsafe { &*cell.get_ptr() }
     }
 
-    fn get_cell(&'py self) -> &'py PyCell<T> {
+    pub(crate) fn get_cell(&'py self) -> &'py PyCell<T> {
```

Comment:
Does this need to be `&'py self` or could it be `&self`?

Reply:
Ugh, so it could _probably_ just be `&self` and things would work but the problem is that because `PyCell<T>` is a GIL Ref, I want to respect that still and keep the `'py` very formal for now. This is exactly part of what I mean by the implementation is a bit messy.

I think that to solve this we might need to split `PyCell` into the public GIL Ref type, and a private inner `PyCell` (or will a different name like `PyCellObject`) and change all this stuff to work off the private inner type. But I'm not completely decided on that.